test/unit/metrics/test_metricframe_process_features.py

```diff
@@ -0,0 +1,168 @@
+# Copyright (c) Microsoft Corporation and Fairlearn contributors.
+# Licensed under the MIT License.
+
+import numpy as np
+import pandas as pd
+import pytest
+
+import fairlearn.metrics as metrics
+from .utils import _get_raw_MetricFrame
+
+
```

Comment:
i would prefer if we were able to write down these tests as tests of the **external** API, rather than an internal API, which i don't expect to be stable. but for now, happy to keep things as is.

Reply:
The cases are also covered through external tests, but this gives an easier way of doing them systematically. Since we only really have one entry point (the constructor), lots can go wrong in a test of that, whereas this lets us target specific portions of it. I believe that `_process_features` could technically be moved out of `MetricFrame` and made a 'module private' rather than 'class private' if that's a particular concern.

Reply:
I understand where this comes from. Just flagging, but no need to address.